DataFormats/Detectors/TRD/include/DataFormatsTRD/EventRecord.h

```diff
@@ -18,6 +18,7 @@
 #include "CommonDataFormat/RangeReference.h"
 #include "FairLogger.h"
 #include "DataFormatsTRD/Tracklet64.h"
+#include "Framework/ProcessingContext.h"
```

Comment:
It is not a good idea to expose framework headers to classes which generate root dictionary. 
This EventRecord is not really a data format which you store or exchange between the devices, it is just a helper class used in the decoding, right? I think it should go to the `O2/Detectors/TRD/reconstruction` and more important, don't generate dictionary for it (remove ClassDefNV, which is anyway wrong since you neither add this header to `o2_target_root_dictionary(DataFormatsTRD` in the DataFormats/Detectors/TRD/CMakeLists.txt nor add these classes to LinkDef.

Reply:
The classdef was there as i originally intended just to serialise it and send it. Will move it to reconstruction. Should I forward declare the processing context as well?

Reply:
Yes, thanks, the less you expose framework headers, the better it is. You forgot to remove the ClassDefNV from EventRecord.h

Reply:
my goodness, i am jumping around branches and took it out of the wrong one. fixing now.